### PR TITLE
UCP/AM: Fixed single vs multi selection adding reply footer size.

### PR DESCRIPTION
**Pull request by rakhmets**
## What
Fixed single vs multi protocol selection taking into account reply footer size.

### CODE REVIEW

File: src/ucp/am/eager_single.c [hunk 1]
```diff
@@ -23,6 +23,12 @@ ucp_am_eager_fill_reply_footer(ucp_am_reply_ftr_t *ftr, ucp_request_t *req)
     ftr->ep_id = ucp_send_request_get_ep_remote_id(req);
 }
 
+static size_t ucp_am_eager_single_hdr_size(ucp_operation_id_t op_id)
+{
+    return op_id == UCP_OP_ID_AM_SEND ? sizeof(ucp_am_hdr_t) :
+            sizeof(ucp_am_hdr_t) + sizeof(ucp_am_reply_ftr_t);
```
Comment (hoopoepg) [inline src/ucp/am/eager_single.c:29]: minor: may be
```
return sizeof(ucp_am_hdr_t) + (op_id == UCP_OP_ID_AM_SEND ? 0 : sizeof(ucp_am_reply_ftr_t));
```

@brminich what do you think?

**Review by hoopoepg — DISMISSED** [review]

File: src/ucp/am/eager_single.c [hunk 2]
```diff
@@ -23,6 +23,12 @@ ucp_am_eager_fill_reply_footer(ucp_am_reply_ftr_t *ftr, ucp_request_t *req)
     ftr->ep_id = ucp_send_request_get_ep_remote_id(req);
 }
 
+static size_t ucp_am_eager_single_hdr_size(ucp_operation_id_t op_id)
+{
+    return sizeof(ucp_am_hdr_t) +
+           (op_id == UCP_OP_ID_AM_SEND ? 0 : sizeof(ucp_am_reply_ftr_t));
```
Comment (brminich) [inline src/ucp/am/eager_single.c:29]:
```suggestion
           ((op_id == UCP_OP_ID_AM_SEND) ? 0 : sizeof(ucp_am_reply_ftr_t));
```

File: src/ucp/am/eager_single.c [hunk 3]
```diff
@@ -84,7 +90,8 @@ ucp_am_eager_short_proto_init_common(const ucp_proto_init_params_t *init_params,
                                      size_t min_iov, ucp_operation_id_t op_id)
 {
     const ucp_proto_select_param_t *select_param = init_params->select_param;
-    ucp_proto_single_init_params_t params        = {
+    size_t header_size = ucp_am_eager_single_hdr_size(op_id);
```
Comment (brminich) [inline src/ucp/am/eager_single.c:93]: do u need this var to avoid compiler warning?

**Review by brminich — DISMISSED** [review]

File: test/gtest/ucp/test_ucp_am.cc [hunk 4]
```diff
@@ -847,6 +847,72 @@ UCS_TEST_P(test_ucp_am_nbx, rx_am_mpools,
 UCP_INSTANTIATE_TEST_CASE(test_ucp_am_nbx)
 
 
+class test_ucp_am_nbx_single_multi_reply : public test_ucp_am_nbx {
+public:
+    test_ucp_am_nbx_single_multi_reply()
+    {
+        modify_config("RNDV_THRESH", "inf");
+        modify_config("PROTO_ENABLE", "y");
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_am.cc:855]: run with both proto enable and default (variants)

**Review by yosefe — APPROVED** [review]

**Review by yosefe — APPROVED** [review]

**Pull request merged**